### PR TITLE
build: quality of life improvements

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [x] 🚩 Other

## Description

Quality of life for yarn illiterate people:
- updated all deps that weren't breaking changes or unreleased betas
- separated concerns between dev and prod
- updated React
- removed unused packages to trim install size
- removed all manual enforced tree dependencies

## Related Tickets & Documents

Needed this for #950 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

Need @bdougie to check if I didn't remove anything that was hardcoded because of deployment - should be addressed to remove concern.

### CODE REVIEW

**Comment by 0-vortex** [comment]
can drop the matrix for node 12 and add 15 too

File: src/index.js [hunk 1]
```diff
@@ -5,13 +5,13 @@ import Config from "./config";
 import {getUserFromJwt} from "./lib/identityActions";
 import "./index.css";
 import registerServiceWorker from "./registerServiceWorker";
-import OneGraphApolloClient from "onegraph-apollo-client";
```
Comment (bdougie) [inline src/index.js:8]: OneGraphApollo Client uses AuthGuardian library, which means it may need to stay.

Comment (bdougie) [reply src/index.js:8]: I think I am wrong about this one. Disregard.

**Comment by filiptronicek** [comment]
> 
> 
> can drop the matrix for node 12 and add 15 too

@0-vortex I think it's still worth running the jobs on Node v12. I agree Node v15 would be a great addition (maybe even add Node v16?)

**Comment by mtfoley** [comment]
@0-vortex After the stream from yesterday, I tried `nvm install v15`, `npm ci` on the same clone I had where I replicated the prismjs errors. For me the commands succeeded and the tests passed. 

I got basically the same results for node@14.16.1, npm@7.13.0

@bdougie Perhaps in the CI workflow it'd make sense to put a line `npm install -g npm@latest` prior to the `npm ci` command.

I'm going to look into these packages that got flagged as deprecated.

```
➜  open-sauced git:(build-qol) nvm install v15
Downloading and installing node v15.14.0...
Downloading https://nodejs.org/dist/v15.14.0/node-v15.14.0-darwin-x64.tar.xz...
########################################################################################################################################################################################## 100.0%
Computing checksum with shasum -a 256
Checksums matched!
Now using node v15.14.0 (npm v7.7.6)
➜  open-sauced git:(build-qol) git status                                                                                                     
On branch build-qol
Your branch is up to date with 'origin/build-qol'.

nothing to commit, working tree clean
➜  open-sauced git:(build-qol) npm ci
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'

added 3884 packages, and audited 4318 packages in 11m

286 packages are looking for funding
  run `npm fund` for details

114 vulnerabilities (5 low, 98 moderate, 10 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

Running `npm test` after that, all passes.

**Comment by 0-vortex** [comment]
> > can drop the matrix for node 12 and add 15 too
> 
> @0-vortex I think it's still worth running the jobs on Node v12. I agree Node v15 would be a great addition (maybe even add Node v16?)

think my previous comments were a bit vague - essentially I view open sauced as a 2 fragment project: the react client and the graphql module that is currently baked into the client logic. Eventually splitting the project into 2 specific npm modules, `open-sauced` client app (private, statically built for current infra) and `open-sauced-graphql` which would be a re-usable npm package (public, matrix build for any infra)

failed to previously mention "why" current matrix is irrelevant but re-usable for any module split. the real life use case simplifies things a lot since `open-sauced-graphql` would be used in multiple projects built on different platforms and requirements, it is required to never add a shrinkwrap to that project (the project still gets lockfile-life security updates but resolution is lax) and then `open-sauced` itself would have some value for building on multiple operating systems, ensuring everyone can run the project locally and contribute, but from a lockfile perspective `package.engines` would specify the versions you need to set up your version manager to set up node and npm - with those versions installed locally and a proper lockfile, subsequent changes that do not affect versioned dependencies would not trigger a lockfile change in the case of a simple fresh install `npm install` as it does now (this has other client delays that documentation or onboarding videos would not solve)

been trying to solve all of the above without actually shrinkwrapping and started a rabbit hole :D

2 commits away from magic tho 🗡️

**Comment by 0-vortex** [comment]
got you guys covered but have to reopen this after that force push

**Comment by filiptronicek** [comment]
This PR has been moved to https://github.com/open-sauced/open-sauced/pull/1011

**Pull request closed without merging**